benchmarks/src/main/java/io/druid/benchmark/FilterPartitionBenchmark.java

```diff
@@ -492,68 +453,54 @@ public void readComplexOrFilterCNF(Blackhole blackhole) throws Exception
   }
 
-  private Sequence<List<String>> readCursors(Sequence<Cursor> cursors, final Blackhole blackhole)
+  private void readCursors(Sequence<Cursor> cursors, Blackhole blackhole)
   {
-    return Sequences.map(
+    final Sequence<Void> voids = Sequences.map(
         cursors,
-        new Function<Cursor, List<String>>()
-        {
-          @Override
-          public List<String> apply(Cursor input)
-          {
-            List<String> strings = new ArrayList<String>();
-            List<DimensionSelector> selectors = new ArrayList<>();
-            selectors.add(
-                input.getColumnSelectorFactory().makeDimensionSelector(new DefaultDimensionSpec("dimSequential", null))
-            );
-            //selectors.add(input.makeDimensionSelector(new DefaultDimensionSpec("dimB", null)));
-            while (!input.isDone()) {
-              for (DimensionSelector selector : selectors) {
-                IndexedInts row = selector.getRow();
-                blackhole.consume(selector.lookupName(row.get(0)));
-                //strings.add(selector.lookupName(row.get(0)));
-              }
-              input.advance();
+        input -> {
+          List<DimensionSelector> selectors = new ArrayList<>();
+          selectors.add(
+              input.getColumnSelectorFactory().makeDimensionSelector(new DefaultDimensionSpec("dimSequential", null))
+          );
+          while (!input.isDone()) {
+            for (DimensionSelector selector : selectors) {
+              IndexedInts row = selector.getRow();
+              blackhole.consume(selector.lookupName(row.get(0)));
             }
-            return strings;
+            input.advance();
           }
+          return null;
         }
     );
+
+    blackhole.consume(voids.toList());
   }
 
-  private Sequence<List<Long>> readCursorsLong(Sequence<Cursor> cursors, final Blackhole blackhole)
+  private void readCursorsLong(Sequence<Cursor> cursors, final Blackhole blackhole)
   {
-    return Sequences.map(
+    final Sequence<Void> voids = Sequences.map(
         cursors,
-        new Function<Cursor, List<Long>>()
-        {
-          @Override
-          public List<Long> apply(Cursor input)
-          {
-            List<Long> longvals = new ArrayList<Long>();
-            BaseLongColumnValueSelector selector = input.getColumnSelectorFactory().makeColumnValueSelector("sumLongSequential");
-            while (!input.isDone()) {
-              long rowval = selector.getLong();
-              blackhole.consume(rowval);
-              input.advance();
-            }
-            return longvals;
+        input -> {
+          BaseLongColumnValueSelector selector = input.getColumnSelectorFactory()
+                                                      .makeColumnValueSelector("sumLongSequential");
+          while (!input.isDone()) {
+            long rowval = selector.getLong();
+            blackhole.consume(rowval);
```

Comment:
I suggest to use some real aggregator, like LongSum, to make the load more realistic

Reply:
Do you think it matters as long as the load is consistent between tests? I'd rather leave it using blackhole (especially since the string one can't switch to an aggregator anyway)